### PR TITLE
sample: dfu: allow sample to be build on default platform

### PR DESCRIPTION
**Pull request by jfischer-no**
Remove platform_allow and allow sample to be build on
default platform but not on native posix.

### CODE REVIEW

**Comment by de-nordic** [comment]
You probably want to exclude all platforms that do not define `image_0` partition.

**Comment by carlescufi** [comment]
@de-nordic and @PerMac could you please take a look?

**Comment by de-nordic** [comment]
@jfischer-no seems that the builtkite problems, at least with the mimxrt1064_evk, have existed prior to the PR but have been uncovered here.
I have tried to build the sample from master (f0352500b3e5c3aaecca17d03b1e07bb2e0ac973) and found that I can not even go through configuration phase to menuconfig.
Question is whether the fix for these problems should be part of this PR, or the problems should be filtered out and fixed as part of other PR.

**Comment by de-nordic** [comment]
@jfischer-no Is this still relevant?

**Review by de-nordic — APPROVED** [review]
Looks OK

**Comment by de-nordic** [comment]
@PerMac Can you also look at this?

**Comment by PerMac** [comment]
Looks good. Just please change `sanitycheck:` to `twister:` in the commit title as there is no more "sanitycheck" :)

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@galak and @mbolivar-nordic please take a look

**Review by nashif — APPROVED** [review]

File: scripts/pylib/twister/expr_parser.py [hunk 1]
```diff
@@ -264,6 +264,13 @@ def ast_expr(ast, env, edt):
 
         return ast_handle_dt_enabled_alias_with_parent_compat(edt, alias,
                                                               compat)
+    elif ast[0] == "dt_compat_enabled_with_label":
+        compat = ast[1][0]
+        label = ast[1][1]
+        for node in edt.nodes:
+            if node.status == "okay" and label in node.labels and node.matching_compat == compat:
+                return True
+        return False
```
Comment (mbolivar-nordic) [inline scripts/pylib/twister/expr_parser.py:273]: I'm late to the party, but this should use `edt.label2node` instead for speed. Something like this (untested):

```py
node = edt.label2node.get(label)
return node is not None and node.status == 'okay' and node.matching_compat == compat
```

@jfischer-no can you send a follow-up?

Comment (jfischer-no) [reply scripts/pylib/twister/expr_parser.py:273]: @mbolivar-nordic thanks for the hint, see #33387

**Pull request merged**